llvm/lib/CheerpWriter/PartialExecuter.cpp

```diff
@@ -1727,6 +1727,32 @@ void PartialInterpreter::visitOuter(FunctionData& data, llvm::Instruction& I)
 			return;
 		}
 	}
+	if (const BinaryOperator* BO = dyn_cast<BinaryOperator>(&I))
```

Comment:
I think there is a better place for this logic, there are instruction based checks somewhere else. Cannot check right now, but see for ifs over opcodes or instruction types